modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/FreeMarkerTool.java

```diff
@@ -369,7 +371,9 @@ public String getGraphQLPropertyName(
 				Map<String, Schema> propertySchemas =
 					schema.getPropertySchemas();
 
-				if (propertySchemas != null) {
+				if (!StringUtil.equalsIgnoreCase(schemaName, entry.getKey()) &&
```

Comment:
Can you use entry.getKey() and compare it with the propertyName inside _isGraphQLPropertyRelation instead of inferring it by removing the id?